apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -37,9 +38,60 @@ const styles = {
 
 class TeacherFeedback extends Component {
   static propTypes = {
-    viewAs: PropTypes.oneOf(['Teacher', 'Student']),
     //temp prop for which version to display (stable, released 2018-teacher-experience, or internal, developer version)
-    withUnreleasedFeatures: PropTypes.bool
+    withUnreleasedFeatures: PropTypes.bool,
+
+    //Provided by Redux
+    viewAs: PropTypes.oneOf(['Teacher', 'Student']),
+    serverLevelId: PropTypes.number,
+    teacher: PropTypes.number,
+  };
+
+  constructor(props) {
+    super(props);
+    const search = window.location.search;
+    const studentId = search.split('&')[1].split("=")[1];
+
+    this.state = {
+      comment: "",
+      studentId: studentId,
+      latestFeedback: [],
+    };
+  }
+
+  componentDidMount = () => {
+    $.ajax({
+      url: '/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id='+this.state.studentId+'&level_id='+this.props.serverLevelId+'&teacher_id='+this.props.teacher,
+      method: 'GET',
+      contentType: 'application/json;charset=UTF-8',
+    }).done(data => {
+      this.setState({latestFeedback: [data]});
+    });
+  };
+
+  onCommentChange = (event) => {
+    this.setState({comment: event.target.value});
+  };
+
+  onSubmitFeedback = () => {
+    const payload = {
+      comment: this.state.comment,
+      student_id: this.state.studentId,
+      level_id: this.props.serverLevelId,
+      teacher_id: this.props.teacher
+    };
+
+    $.ajax({
+      url: '/api/v1/teacher_feedbacks',
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      dataType: 'json',
+      data: JSON.stringify({teacher_feedback: payload})
+    }).done(data => {
+      this.setState({latestFeedback: [data]});
+    }).fail((jqXhr, status) => {
+      console.log(status + "  " + jqXhr.responseJSON);
```

Comment:
Let's do more robust error handling here before we ship (fine while behind a flag though).

Reply:
https://github.com/code-dot-org/code-dot-org/pull/23441